src/collection-view.js

```diff
@@ -125,11 +125,89 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, options) {
+    const changes = options.changes;
+    if (changes) {
```

Comment:
turns out there's no need to check for changes here.. it's always available when `update` is triggered.